api/v1alpha1/hosted_controlplane.go

```diff
@@ -102,6 +102,17 @@ type HostedControlPlaneSpec struct {
 	// provided: reconciliation is paused on the resource until the field is removed.
 	// +optional
 	PausedUntil *string `json:"pausedUntil,omitempty"`
+
+	// OLMCatalogPlacement specifies the placement of OLM catalog components. By default,
+	// this is set to management and OLM catalog components are deployed onto the management
+	// cluster. If set to guest, the OLM catalog components will be deployed onto the guest
+	// cluster.
+	//
+	// +kubebuilder:validation:Enum=management;guest
+	// +kubebuilder:default=management
+	// +optional
+	// +immutable
+	OLMCatalogPlacement string `json:"olmCatalogPlacement,omitempty"`
```

Comment:
can we make this an enum type with constant values?

Reply:
Oh I was under the impression that enum is being set, https://github.com/hasueki/hypershift/blob/72a22597795267c52ab8ba46a3d7af7e3e792030/api/v1alpha1/hostedcluster_types.go#L235. Or are you referring to something else?

Reply:
I meant defining a separate type for it and constants for that type as in
https://github.com/openshift/hypershift/blob/main/api/v1alpha1/hostedcluster_types.go#L432-L452

Reply:
Gotcha. Yup I can update.